plugins/inventory/nmap.py

```diff
@@ -9,7 +9,7 @@
     name: nmap
     short_description: Uses nmap to find hosts to target
     description:
-        - Uses a YAML configuration file with a valid YAML extension.
+        - Uses a YAML configuration file that ends with C(nmap.(yml|yaml)).
```

Comment:
That would be a breaking change. What's the reasoning for this?

I would move this into a separate PR, since this needs special handling. We generally do not accept breaking changes without good reasoning, or at least a deprecation period.

Reply:
Honestly, the only reason for me changing that in the first place (two years ago) was because the developer guide for dynamic inventory plugins stated that it should be a requirement and the plugin I used as a "guideline" was the virtualbox plugin which requires vbox as a file extension.
I hesitated to include this now and I agree that it should be a separate PR since it has nothing to do with the cache fix.